### ISSUE TITLE
Feature: Add (or expose) tags for Baby diapers, Period products, and similar properties… to the org listings, search, and edit screens

### ISSUE THREAD

**Issue by fancyham** [hackforla/food-oasis#2115]
**Dependency**
- [x] #947 
- [ ]  Determine what new tags are high priority items for our users — Diapers, Period products, LGBTQ+ -friendly, child-friendly, etc., then create these tags using the existing ‘custom tags’ database field
- [ ]  Actually tag our data (this will require volunteers asking orgs for these details and updating records accordingly)

This is priority 3/5 (where 5 is most urgent).  This will help a lot of families, and sets the stage for much better search functionality.

This could involve a developer and a product/designer/researcher to help with how it looks and for placement of checkboxes on the edit screens.  If the latter is not available, use your best judgement as it can be modified later.

Research is also useful to determine what other high-priority tags we'd want to add.

### Overview

Enable searching and displaying (new?) tags for our listings: Baby diapers, Period products, Kosher, Halal

This may involve adding checkboxes for each tag to the edit interface, displaying these tags on search results, and adding these tags to the filters available to food seekers.

### Context

Right now we have minimal 'tagging' of our content — pantry and/or meal program, what types of pantry items are available (baked goods, dairy, etc.).  But there are other important criteria that people search on.

Other sites support this feature and it can be pretty important when homing in on locations to contact or visit.

For example, some seekers are also looking for diapers and period products.  Since most pantries do not provide these, finding one can make a big difference to folks.  Also, like food, both are expensive and are necessities.

Fortunately, some (but not many) other listing websites now are starting to highlight when these are available.  

We should add these tags because these are important for food seekers and because it reduces the search effort for our users.

### How is it now?
Currently we don't have a way to filter or search by any tags.  Also, our text descriptions are not yet searchable, so no way to find a listing that mentioned 'baby diapers' or 'kosher'.  This means the user must read every single listing and call, if they want to find out if the service meets their criteria.

**John D said that we currently use a tags-type system already so perhaps we can expose that existing system to the user.**

### Proposed new feature
**The goal would be to highlight these tags, if present, in the listing:**
<img width="300" alt="image" src="https://github.com/hackforla/food-oasis/assets/3376957/86adfe23-5880-4e56-a278-487bc855c428">

**And to use these tags as filters:**
<img width="200" alt="image" src="https://github.com/hackforla/food-oasis/assets/3376957/57ef5390-063b-4ef7-8376-e9d27743234d">

**And add checkboxes to the organization edit interface (if needed):**
<img width="640" alt="image" src="https://github.com/hackforla/food-oasis/assets/3376957/293d0878-7a0c-45ab-8447-231ae6d51dc5">

To start, we’d want these tags (or their equivalent): 
- Baby Diapers, 
- Period products, 
- Kosher
- Halal

In the future, pending more research and review of our ability to be accurate, we may want to add more tags such as these:
- LGBTQ+-friendly tags as well (as this is also important (for example LGBTQ+ students [need to know what's a safe space](https://www.foodsystemsjournal.org/index.php/fsj/article/download/1146/1119/2312) before visiting a pantry))
- Eligibility requirements or "No eligibility requirements" (need to do more research about what those classifications might be, or if this is even possible)
- "Same-day", "Walk-Up"
- "Sermon required" (some meal programs require you sit for a sermon before receiving food)
- Baby formula

Goals:
- [ ] 1. Add new tags onto the organization editing screens (if needed)
- [ ] 2. Allow users to search/filter by these tags
- [ ] 3. Highlight these tags in search results and detailed listings

### References:
Examples from other sites:
https://www.lafoodbank.org/find-food/pantry-locator/ 
![image](https://github.com/hackforla/food-oasis/assets/3376957/2e510ec4-2762-4b96-bc48-2640649f3f64)

Article: [Should Period Products Be Free?](https://www.nytimes.com/2022/09/28/learning/should-period-products-be-free.html?ugrp=c&unlocked_article_code=1.j00.kmfb.roWj7aBzihtb&smid=url-share)
Article: [Three items you didn’t know some food banks distribute](https://www.feedingamerica.org/hunger-blog/three-non-food-items-food)

**Comment by fancyham** [comment]
If you have additional potential tags, now’s a good time to discuss and possibly list them in priority order by need (which will then need to be evaluated for ’can we actually get this information reliably?)).  There’s a list of possible future tags in the issue so feel free to modify it.

Also, if you have past research or SME comments about these types of things, please add it to the description in the ‘references’ section and re-sort as needed.

**Comment by SAUMILDHANKAR** [comment]
1. Progress: Looked at the codebase a little bit but didn't make much progress last week.
2. Blockers: No blockers.
3. Availability: 5 hours before next dev meeting. Thursday 7-9 am IST, Saturday 7-8 am IST, Monday 7-9 am IST.
4. ETA: Nov 5, 2024

**Comment by fancyham** [comment]
This is hard to do because we don’t have tags like this in the database right now for things like diapers, period products, etc.

So the blockers on this are not programmatic issues but rather process:

- [ ] First, determine what new tags are high priority items for our users — Diapers, Period products, LGBTQ+ -friendly, child-friendly, etc. — then decide to start tagging our data accordingly.  John D says there is already a database field that allows us to create custom tags.  We’d use that.
- [ ] Actually tag our data (this will require volunteers asking orgs for these details and updating records accordingly)
- [ ] Then this issue comes into play.  We build the UI using those tags.

In the meantime, the goal is still to get people access to these types of listings.  We can actually do some of that now based on information currently stored in description fields and elsewhere.

So instead. of this, I suggest we do/prioritize the next-best thing instead: full-text search.  

That would be a meaningful and useful half-step forward.  It also means we can search our own data more easily.

See this issue for more details:
- #2117

**Comment by TifNi** [comment]
@fancyham @fischeme @entrotech 
- Product agrees with Mark that we shouldn't build or maintain a complex taxonomy.
- Some tagging is warranted... but which option we use to go about that will be discussed later, after #947 is completed
   - Possible addition of a new category for e.g. pantry, meal, personal items
   - Or, tags for just the following:
      - Baby
      - Period
      - Cleaning
- LGBTQ friendly, Halal and Kosher info should be in the free-text field
- If Sermon are required in order to receive, that should be classified under the requirements

**Comment by fancyham** [comment]
@Luna4423-arch — please add your thoughts about tags for take-away and dine-in tags here in the comments

**Comment by fancyham** [comment]
- This is not dependent on #947 being complete.  If it is, please list out specifically what this is depending on that doesn’t currently exist.

**Comment by SAUMILDHANKAR** [comment]
> * This is not dependent on [Epic: Advanced filters for search #947](https://github.com/hackforla/food-oasis/issues/947) being complete.  If it is, please list out specifically what this is depending on that doesn’t currently exist.

@fancyham So, I added the dependency and unfortunately cannot trace the discussion which led to this decision. So, feel free to ignore my comment and do as deemed best. However, fwiw let me add my memory dump. So the following two points mentioned in #947 
- What types of filters do our users want? (for example, open on certain days of the week, religious or not, vegetarian, shelter or not, etc.)
- What’s feasible technically? What’s easy, what’s hard?
is probably why the dependency was added. Meaning, the idea was let's decide what all we want through research and check the technical implementation feasibility of it. Since those action items are unchecked I am assuming that part is still pending or there might be work being done elsewhere. That is all I can add.

**Comment by fancyham** [comment]
@Luna4423-arch (please add details) is reading research papers to figure out what would be top priority items for our food-seekers.  Need to limit how many because of difficulty of collecting info.  Want to keep it MVP and simple for now.

**Comment by Luna4423-arch** [comment]
# Research Summary: Why Menstrual Products Are Important

Access to menstrual products is an important health and equity issue, yet many people in the United States struggle to afford them. Research shows that about 42% of menstruators have experienced difficulty buying period products, contributing to a widespread issue known as period poverty (Glayzer et al., 2024). This lack of access can affect health, school participation, and work stability.

Community solutions such as period product pantries help address immediate needs, but broader policy changes are still required to solve the issue long term (Glayzer et al., 2024).

For example, The River Food Pantry has reported increasing demand for pads, tampons, and panty liners due to rising living costs. The pantry relies on donations and financial support to meet growing community needs, showing how local organizations are filling gaps in public assistance systems.

At the policy level, a U.S. Government Accountability Office (GAO) report found that menstrual products are not a required Medicaid benefit. States are not obligated to provide them, although they can choose to offer coverage through managed care plans or special programs (U.S. GAO, 2024).

The GAO found that 25 states have at least one Medicaid managed care plan that offers some form of menstrual product coverage. However, access is inconsistent because coverage depends on the specific plan, not the state as a whole. Some plans offer direct menstrual product benefits, while others include them under broader over-the-counter or personal care allowances (U.S. GAO, 2024).

## Insight

I think menstrual products should be a top priority because they are essential for daily life, and access to them helps women work, attend school, and go out in public with dignity and confidence.

## References

- Glayzer, E. J., et al. (2024). *Fighting for menstrual equity through period product pantries.* Women's Health, 20, 1–11. https://doi.org/10.1177/17455057241281459  
- U.S. Government Accountability Office. (2024). *Medicaid: Menstrual product coverage (GAO-24-106960).* https://www.gao.gov

**Comment by Luna4423-arch** [comment]
# Beyond Food: Why Menstrual, Baby, and Hygiene Products Belong in Food Assistance

Food insecurity is not just about food. Families who struggle to afford food may also struggle to afford diapers, menstrual products, and basic hygiene items. However, these products are usually not covered by SNAP and are not always available at food pantries.

## Menstrual Products

Many low-income women who experience food insecurity also struggle to afford menstrual products. One survey found that 64% face this challenge.

Some food pantry users have said they would like to receive period products along with food. Providing these products can help families meet their basic needs.

**Sources:**
- Brookings — [Period poverty and its reach across the US](https://www.brookings.edu/articles/period-poverty-and-its-reach-across-the-us/)
- PMC/NIH — [Editorial: Period poverty](https://pmc.ncbi.nlm.nih.gov/articles/PMC9941660/)
- Period.org — [Period Poverty overview](https://period.org/periodpoverty)

## Baby and Diaper Products

Diapers are another important expense for families with young children. The LA Regional Food Bank has distributed more than 4.3 million diapers since 2019.

Research shows that families who struggle to afford diapers are also more likely to experience food insecurity. Since SNAP does not cover diapers, providing them through food assistance programs can help families save money for food.

**Sources:**
- LA Regional Food Bank — [Diaper Program](https://www.lafoodbank.org/stories/diaper-program/)
- PMC/NIH — [Diaper need and food insecurity, WIC study](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7930593/)

## Hygiene Products

Many low-income families also struggle to afford basic hygiene products such as soap, shampoo, toothpaste, and laundry detergent.

SNAP does not cover these products, so families may have to choose between buying food and buying hygiene supplies. One survey found that 58% of households spent less on food to afford hygiene products.

**Sources:**
- Simply the Basics — [Our Purpose / hygiene insecurity](https://www.simplythebasics.org/purpose)
- Good360 — [The Hidden Costs of Hygiene Poverty](https://good360.org/blog-posts/the-hidden-costs-of-hygiene-poverty/)
## Why This Matters

Food is only one part of household survival. When families cannot afford diapers, menstrual products, or basic hygiene supplies, they may have to make difficult trade-offs between essential needs.

For food assistance organizations, offering these products alongside food can help address the **broader needs of food-insecure communities—not just hunger, but the everyday necessities that allow people to live with health, dignity, and stability.**

**Comment by Luna4423-arch** [comment]
# UI for food seekers and database people

<img width="1200" height="800" alt="Image" src="https://github.com/user-attachments/assets/2937a7d4-2496-4c4d-bc14-1862e0058cad" />

I moved the Confirm button to the bottom of the section because this form is likely to grow — more item categories may be added over time. Placing Confirm at the end means users won't have to scroll back up to submit once they're done,

**Comment by fancyham** [comment]
Talked to research team (who have been doing data validation and administration) and they said that it's okay to leave the 'confirm' button in the top corner where it is for now.

The checkbox is actually just to make sure that the data validator has looked at a data field - but it's always been ambiguous about what the 'data field' is on this page — most of the time, data validators assumed it was the top checkboxes only... but that was all based on our instructions for how to validate data.

Separately, @GigiUxR is going to create two separate new issues (which are not part of this issue):
1) to look at a better arrangement for the sections on this page (because the Eligibility and Requirements field is really important but doesn't look like it needs to be confirmed)
2) to change the word 'confirm' to 'confirmed' or something more descriptive like 'everything on this page has been confirmed'
